addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/UserControlledOpenRedirectScanner.java

```diff
@@ -114,12 +114,12 @@ private void checkUserControllableLocationHeaderValue(
                 continue;
             }
 
-            if (paramValue.equalsIgnoreCase(protocol)
-                    || paramValue.equalsIgnoreCase(domain)
-                    || paramValue.equalsIgnoreCase(token)
+            if (paramValue.equalsIgnoreCase(domain)
                     || (responseLocation.indexOf("://") > 0
                             && paramValue.indexOf(responseLocation) >= 0)) {
-                raiseAlert(msg, id, param.getName(), paramValue, responseLocation);
+                if (requestDomain == null || !requestDomain.equalsIgnoreCase(domain)) {
```

Comment:
This check should be done earlier before the loop (no need to actually check the parameters if it's the same).

Reply:
Makes sense. updated the code as suggested. Please review.